### ISSUE TITLE
`no-nested-ternary` does not autofix nested ternary

### ISSUE THREAD

**Issue by mvorisek** [sindresorhus/eslint-plugin-unicorn#2008]
```js
var cellText = isYear
    ? i
    : isMonth
        ? settings.text.monthsShort[i]
        : isDay
            ? cellDate.getDate()
            : module.helper.dateFormat(formatter.cellTime, cellDate);
```

When the rule can autofix ternary (add parenthesis around), I would expect it to fix any level of nesting.

Currently it seems it can autofix only 2 nested ternary operators.

**Comment by fisker** [comment]
Maybe we should document that.

**Comment by kaushalji-123** [comment]
yeah you are right @mvorisek it can autofix only 2 nested ternary operators. 

`var cellText = isYear
    ? i
    : (isMonth
        ? settings.text.monthsShort[i]
        : (isDay
            ? cellDate.getDate()
            : module.helper.dateFormat(formatter.cellTime, cellDate)));
`

**Comment by sindresorhus** [comment]
I think this should be closed as documented

The fixer only handles the case where a single nested ternary is valid but missing parentheses. For deeper nesting, adding parentheses does not fix the violation, as this rule intentionally disallows nesting beyond one level.

Fixing the reported example would require rewriting it to `if`/`else` or introducing variables, which is too large and opinionated for an autofix. The docs already state that the rule is only fixable when nesting is up to one level.
